modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -224,12 +271,95 @@ public static function validateForm(array &$element, FormStateInterface $form_st
       $form_state->setErrorByName($code_path, t('Coupon is invalid'));
       return;
     }
-    if (!$coupon->getPromotion()->applies($order)) {
-      $form_state->setErrorByName($code_path, t('Coupon is invalid'));
-      return;
-    }
```

Comment:
Howcome this was removed? At least in multiple coupon mode, this makes it appear to apply a coupon (show a success message) even when the coupon conditions don't match.